### ISSUE TITLE
annual temporal weighting

### ISSUE THREAD

**Issue by bradyrx** [bradyrx/esm_analysis#49]
We should add a function that computes annual means, weighted by the length of each month (instead of just doing `ds.mean('time')` to get annual averages).

It's a long process to do every time, as in http://xarray.pydata.org/en/stable/examples/monthly-means.html. Would be nice to just add that as a function.

**Comment by aaronspring** [comment]
I use these, naming as the CDOs
```python
def yearmonmean(ds):
    return ds.groupby('time.year').mean('time').rename({'year': 'time'})

def yearsum(ds):
    return ds.groupby('time.year').sum('time').rename({'year': 'time'})

**Comment by bradyrx** [comment]
I use `ds.groupby('time.year').mean('time').rename({'year': 'time'})` too. Does this actually weight them? Was talking to somebody at this ORNL hackathon who doesn't use `xarray` and his concern was it wasn't weighted and treated all months the same. I always thought pandas weighted them too...

I can test tomorrow using the method in the `xarray` docs and then the standard method to see if it's the same.

**Comment by bradyrx** [comment]
@aaronspring , yep my testing shows that the pandas groupby does not weight properly. So if you're going from monthly -> annual, it just clumps all 12 months from say, 1965, into one bin and does an arithmetic average. I'll add a function to replicate http://xarray.pydata.org/en/stable/examples/monthly-means.html.

**Comment by bradyrx** [comment]
Addressed in #50 .
